TestResultSummaryService/routes/getTabularData.js

```diff
@@ -0,0 +1,51 @@
+const { TestResultsDB, ObjectID } = require( '../Database' );
+
+function exceedDate(jdkDate) {
+    return function(element) {
+       return parseInt(element.aggregateInfo[0].benchmarkProduct.slice(-8)) <= parseInt(jdkDate);
+    }
+}
+
+
+module.exports = async ( req, res ) => {
+    const data = [];
+    const db = new TestResultsDB();
+    console.log("Request received: ", req.query.jdkVersion, req.query.jvmType, req.query.jdkDate, req.query.sdkResource);
+    
+    const datas = [];
+    
+    const query = {buildName: {$regex: ".*_" + 'openjdk' + req.query.jdkVersion.substring(1) + "_" + req.query.jvmType + ".*perf_.*"}};
+    
+    // Get list of distinct platforms and benchmarks that match the JDK pipeline
+    const platforms = await db.distinct("buildName", query);
+    const benchmarks = await db.distinct("aggregateInfo.benchmarkName", query);
+
+    console.log("PLATFORMS: ", platforms);
+    console.log("benchmarks:", benchmarks);
+    
+    for (benchmarkIndex in benchmarks) {
+        let benchmarkQuery;
+        // Return all entries that match the current benchmark and platform
+        for (item in platforms) {
+            benchmarkQuery = {$and: [{ buildName: { $regex : platforms[item] }, sdkResource: req.query.sdkResource}, { aggregateInfo: { $elemMatch: { benchmarkName: benchmarks[benchmarkIndex] }}}
+            ] };
+            const result = await db.getData(benchmarkQuery).toArray();
+
+            // Remove all entries whose build date exceeds the chosen date
+            const exceedFilter = result.filter(exceedDate(req.query.jdkDate));
+            // Setting the latest build date from the available dates
+            const latestDate = Math.max.apply(Math, exceedFilter.map(function(o) { return parseInt(o.aggregateInfo[0].benchmarkProduct.slice(-8)); }));
+            // Remove all runs that are not the latest
+            const dateFilter = exceedFilter.filter(entry => parseInt(entry.aggregateInfo[0].benchmarkProduct.slice(-8)) === latestDate);
+            const latest = Math.max.apply(Math, dateFilter.map(function(o) { return o.timestamp; }));
+            // Keep the latest build with the latest timestamp
+            const latestRun = dateFilter.find(function(o){ return o.timestamp == latest; })
+
+            if (latestRun !== undefined) {datas.push( latestRun );}
+        }
+    }
+    // Return the list of unique platforms for column generation
+    const uniquePlatforms = [...new Set(datas.map(item => item.buildName))];
```

Comment:
This might be a bit confusing since we are returning the buildName for platforms. The front end extracts the proper names for platforms. Even though this might reduce processing on the backend, we should still document and maybe change `uniquePlatforms`. Other option would be to extract the names here and send the right data.